### PR TITLE
Corrections des bugs

### PR DESCRIPTION
**Pull request by selligZeittom**
6 bugs corrigés

### CODE REVIEW

File: stack.hpp [hunk 1]
```diff
@@ -15,19 +15,40 @@ class Stack {
...
-        return std::move(storage_[position_--]);
+    T pop()
+    {
+        if(position_ > 0)
+        {
+            return std::move(storage_[--position_]);
+        } else
+        {
+            throw std::out_of_range("Stack is empty");
+            return std::move(storage_[0]);
```
Comment (cm0x4D) [inline stack.hpp]: I am pretty sure that this line will never be executed.

**Comment by cm0x4D** [comment]
Thanks. Still one (bonus) bug is left. Hint: It is a memory leak.

You did not fix every bug in a single commit. It is good practice to keep commits as atomic as possible.

**Comment by selligZeittom** [comment]
fixed the memory leak. also deleted the unused line

File: stack.hpp [hunk 2]
```diff
@@ -15,19 +15,48 @@ class Stack {
...
+        {
+            throw std::out_of_range("Stack is empty");
+        }
+    }
+
+    ~Stack()
+    {
+        if(storage_ != NULL)
+        {
+            delete storage_;
```
Comment (cm0x4D) [inline stack.hpp]: Wrong delete, it is an array...

**Comment by selligZeittom** [comment]
deleted the array now

**Pull request closed without merging**